metrics/engine/engine.go

```diff
@@ -21,13 +21,21 @@ import (
 
 const thresholdsRate = 2 * time.Second
 
+type testRunState struct {
```

Comment:
Hmm why do we need a custom type for this? Can't we just add these 2 fields to the `MetricsEngine` directly? :thinking:

Reply:
btw I remembered that I did a somewhat similar refactoring to the `MetricsEngine` as a part of the PoCs in https://github.com/grafana/k6/pull/2816 :thinking: There I pass `getCurrentTestRunDuration func() time.Duration` as an argument to `evaluateThresholds()`. Maybe it's worth to backport some of these changes here, take a look at [this diff](https://github.com/grafana/k6/pull/2816/files#diff-802af5adfe429d78b1b72096ac7b586c28622dffa6a34024a7e91f5d75c1d077)